view/frontend/web/js/components/steps/ThePaymentStep.vue

```diff
@@ -253,6 +267,16 @@ export default {
           })
       }
     }
+  },
+  watch: {
+    countryWatcher (newCountry) {
+      this.address.country_id = newCountry.value
+      this.regionWatcher = ''
+      this.onCountryChange()
+    },
+    regionWatcher (newRegion) {
+      this.address.region_id = newRegion.value
+    }
```

Comment:
I think you don't need to use watchers here. IMO regular component data is enough.

Reply:
i'm using it b/c track-by attribute where i can provide id of option not working as I expected b/c it's adding whole object where we need only value from this object and watchers can do this for me.